src/Service/PaymentStateDataService.php

```diff
@@ -79,18 +79,27 @@ public function insertPaymentStateData(string $contextToken, string $stateData,
 
         $fields['token'] = $contextToken;
         $fields['statedata'] = json_encode($stateDataArray);
+        $stateData = $this->getPaymentStateDataFromContextToken($contextToken);
 
-        $this->paymentStateDataRepository->create(
-            [$fields],
-            \Shopware\Core\Framework\Context::createDefaultContext()
-        );
+        if ($stateData) {
+            $fields['id'] = $stateData->getId();
+            $this->paymentStateDataRepository->update(
+                [$fields],
+                \Shopware\Core\Framework\Context::createDefaultContext()
+            );
+        } else {
+            $this->paymentStateDataRepository->create(
+                [$fields],
+                \Shopware\Core\Framework\Context::createDefaultContext()
+            );
+        }
```

Comment:
Can we use `EntityRepository#upsert` instead?

Reply:
done!